code/Language/Drasil/SentenceExtract.hs

```diff
@@ -0,0 +1,129 @@
+module Language.Drasil.SentenceExtract (getDoc)where
+
+import Control.Lens ((^.))
+import Language.Drasil.Document
+import Language.Drasil.Spec
+import Language.Drasil.Chunk.AssumpChunk
+import Language.Drasil.Chunk.Change
+import Language.Drasil.Chunk.Eq (QDefinition)
+import Language.Drasil.Chunk.References
+import Language.Drasil.Unit(UnitDefn)
+import Language.Drasil.NounPhrase 
+import Language.Drasil.NounPhrase.Core
+import Language.Drasil.Chunk.Concept
+import Language.Drasil.Chunk.Citation
+import Language.Drasil.Chunk.ReqChunk
+import Language.Drasil.Chunk.Quantity
+import Language.Drasil.Chunk.ShortName
+import Language.Drasil.Classes (HasUID(uid),NamedIdea(term), Idea(getA),
+  HasSymbol(symbol), IsUnit, ExprRelat(relat), HasDerivation(derivations), 
+  HasReference(getReferences), ConceptDomain, Definition(defn))
+import Data.List(transpose, head, tail)
+
+getDoc :: Document -> [Sentence]
+getDoc (Document t a s) = t : a : concatMap getSec s
+
+getSec :: Section -> [Sentence]
+getSec (Section _ _ ("AuxConstants") _) = []
+getSec (Section t sc _ _) = t : concatMap getSecCon sc
+
+getSecCon :: SecCons -> [Sentence]
+getSecCon (Sub s) = getSec s
+getSecCon (Con c) = getCon c
+
+isVar :: ([Sentence], [[Sentence]]) -> [Sentence]
+isVar (((S "Var") : tl), s) = head s
+isVar ((hd : tl, s)) = isVar (tl, tail s)
+isVar ([], _) = []
+isVar (_, []) = []
+
+getCon :: Contents -> [Sentence]
+getCon (Table s1 s2 t _ _) = isVar (s1, transpose s2) --s1 ++ concat s2 ++ [t]
+getCon (Paragraph s) = [s]
+getCon (EqnBlock _ _) = []
+getCon (Definition d) = getDtype d
+getCon (Enumeration lst) = getLT lst
+getCon (Figure l _ _ _) = [l]
+getCon (Requirement reqc) = getReq reqc
+getCon (Assumption assc) = getAss assc
+getCon (Change chg) = getChg chg
+getCon (Bib bref) = getBib bref
+getCon (Graph [(s1, s2)] _ _ l _) = s1 : s2 : [l]
+getCon (Defnt dt (hd:fs) a) = (concatMap getCon $ snd hd) ++ getCon (Defnt dt fs a)
+getCon (Defnt dt [] a) = []
+getCon (_) = []
+
+
+getDtype :: DType -> [Sentence]
+getDtype (Data q) = getQDef q
+getDtype (Theory t) = getTerm t ++ getDefn t
+getDtype _ = []
+
+
+getQDef :: QDefinition -> [Sentence]
+getQDef a = concatMap getRef (a ^. getReferences) ++ (a ^. derivations) ++ getTerm a ++ getUnitD (getUnit a)
+
+getRef :: Reference -> [Sentence]
+getRef (SourceRef s) = [s]
+
+getNP :: NP -> [Sentence]
+getNP (ProperNoun _ _) = []
+getNP (CommonNoun _ _ c) = getCap c
+getNP (Phrase s _ c1 c2) = s : getCap c1 ++ getCap c2
+
+getCap :: CapitalizationRule -> [Sentence]
+getCap (Replace s) = [s]
+getCap (_) = []
+
+getUnitD :: Maybe UnitDefn -> [Sentence]
+getUnitD (Nothing) = []
+getUnitD (Just a) = getTerm a ++ getDefn a
+
+getTerm :: (NamedIdea a) => a -> [Sentence]
+getTerm a  = getNP (a ^. term)
+
+getDefn :: (Definition a) => a -> [Sentence]
+getDefn a = [a ^. defn]
+
+getReq :: ReqChunk -> [Sentence]
+getReq a = [(requires a)]
```

Comment:
`[requires a]`. Please remove all these extra () in your code.

Reply:
Extra parentheses were removed in Document.Extract.hs file in b8706ba42497fa00187eaed1a4613342095c1b14.